### PR TITLE
RDCC-1584: Increasing Mutation Coverage to meet threshold

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1585

### Change description ###

Adding unit tests to increase mutation coverage & small refactoring of code 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java [hunk 1]
```diff
@@ -47,36 +46,31 @@
         return pbaNumbers;
     }
 
-    private  List<String> getPbaNumbersFromPaymentAccount(List<PaymentAccount> paymentAccountsEntity) {
+    private List<String> getPbaNumbersFromPaymentAccount(List<PaymentAccount> paymentAccountsEntity) {
 
         List<String> paymentAccountPbaNumbers = new ArrayList<>();
 
-        if (!paymentAccountsEntity.isEmpty()) {
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java:54]: This line was removed as this check is already taking place in line 31, therefore the mutated condition of this line was untestable/impossible to cover

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java [hunk 2]
```diff
@@ -47,36 +46,31 @@
...
-            );
-        }
         return paymentAccountPbaNumbers;
     }
 
     private List<String> getPbaNumbersFromPaymentAccount(List<PaymentAccount> userMapPaymentAccount, List<PaymentAccount> paymentAccountsEntity) {
 
         List<String> paymentAccountPbaNumbers = new ArrayList<>();
 
-        if (!paymentAccountsEntity.isEmpty()) {
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java:69]: This line was removed as it is redundant - this check is already taking place in line 31, therefore the mutated condition of this line was untestable/impossible to cover

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 3]
```diff
@@ -104,7 +104,7 @@ public OrganisationResponse createOrganisationFrom(
         return new OrganisationResponse(organisation);
     }
 
-    private Organisation saveOrganisation(Organisation organisation) {
+    public Organisation saveOrganisation(Organisation organisation) {
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:107]: These methods were made public as per discussion and agreement with @shashank-rastogi:

> a new instance of Organisation is created in the method and then passed to saveOrganisation method, so I am unable to do a verify on this new Organistion instance as I cannot mock it

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 4]
```diff
@@ -128,7 +128,7 @@ public void addPbaAccountToOrganisation(Set<String> paymentAccounts, Organisatio
         }
     }
 
-    private void addSuperUserToOrganisation(
+    public void addSuperUserToOrganisation(
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:131]: These methods were made public as per discussion and agreement with @shashank-rastogi:

> a new instance of Organisation is created in the method and then passed to saveOrganisation method, so I am unable to do a verify on this new Organistion instance as I cannot mock it

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java [hunk 5]
```diff
@@ -560,18 +566,13 @@ public void findUserStatusByEmailForPendingOrgThrowsException() throws Exception
     }
 
     @Test(expected = ExternalApiException.class)
-    public void findUserStatusByEmailForActiveThrowsExceptionWhenUpServiceDown() throws Exception {
+    public void findUserStatusByEmailForActiveThrowsExceptionWhenUpServiceDown() {
         organisation.setStatus(OrganisationStatus.ACTIVE);
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:571]: prefix test and add verify for when conditions

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java [hunk 6]
```diff
@@ -594,6 +595,8 @@ public void checkUserStatusIsActiveByUserId() {
 
     @Test(expected = AccessDeniedException.class)
     public void checkUserStatusIsActiveByUserId_Throws403_WhenNoUserFoundWithGivenId() {
+        when(professionalUserRepository.findByUserIdentifier(any(String.class))).thenReturn(null);
+
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:599]: prefix test and add verify for when conditions

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java [hunk 7]
```diff
@@ -143,11 +143,16 @@ public void deleteUserAndPaymentAccountsFromUserAccountMapTest() {
 
     @Test
     public void deletePaymentAccountsFromOrganisationTest() {
-        when(organisationRepositoryMock.findByOrganisationIdentifier(any(String.class))).thenReturn(organisation);
+        Organisation organisationMock = mock(Organisation.class);
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:146]: prefix test to the test case

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImplTest.java [hunk 8]
```diff
@@ -460,11 +537,11 @@ private void assertExpectedOrganisationResponse(OrganisationResponse organisatio
         final int orgIdLength = 7;
         assertThat(organisationResponse).isNotNull();
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImplTest.java:538]: prefix test and add verify for when conditions

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java (hunk 5 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:571]: this has been addressed in the following commit: https://github.com/hmcts/rd-professional-api/pull/620/commits/b5448bbd468688f02bf1fd93ebc725364a468785

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java (hunk 7 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:146]: This has been addressed in the following commit: https://github.com/hmcts/rd-professional-api/pull/620/commits/b5448bbd468688f02bf1fd93ebc725364a468785

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java (hunk 6 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java:601]: This has been addressed in the following commit: https://github.com/hmcts/rd-professional-api/pull/620/commits/c268d76aff9fd43600c21a25a63cedcf73515b00

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImplTest.java (hunk 8 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImplTest.java:488]: This has been addressed in the following commit: https://github.com/hmcts/rd-professional-api/pull/620/commits/c268d76aff9fd43600c21a25a63cedcf73515b00

**Pull request merged**